src/pepr/operator/controllers/network/generators/kubeAPI.ts

```diff
@@ -41,82 +56,107 @@ export function kubeAPI() {
   try {
     log.debug(
       "Processing watch for endpointslices, getting k8s service for updating API server CIDR",
     );
     const svc = await K8s(kind.Service).InNamespace("default").Get("kubernetes");
-    await updateAPIServerCIDR(slice, svc);
+    await updateAPIServerCIDR(svc, slice);
   } catch (err) {
     const msg = "Failed to update network policies from endpoint slice watch";
     log.error({ err }, msg);
   }
 }
 
 /**
- * When the kubernetes Service is created or updated, update the API server CIDR
- * @param svc The Service for the API server
+ * When the Kubernetes Service is created or updated, update the API server CIDR.
+ *
+ * If a static CIDR is defined, it skips fetching the EndpointSlice and uses the static value.
+ *
+ * @param {kind.Service} svc - The Service object for the API server.
  */
 export async function updateAPIServerCIDRFromService(svc: kind.Service) {
   try {
-    log.debug(
-      "Processing watch for api service, getting endpoint slices for updating API server CIDR",
-    );
-    const slice = await K8s(kind.EndpointSlice).InNamespace("default").Get("kubernetes");
-    await updateAPIServerCIDR(slice, svc);
+    if (UDSConfig.kubeApiCidr) {
+      log.debug("Processing watch for api service, using configured API CIDR for endpoints");
+      await updateAPIServerCIDR(svc, UDSConfig.kubeApiCidr);
+    } else {
+      log.debug(
+        "Processing watch for api service, getting endpoint slices for updating API server CIDR",
+      );
+      const slice = await K8s(kind.EndpointSlice).InNamespace("default").Get("kubernetes");
+      await updateAPIServerCIDR(svc, slice);
+    }
   } catch (err) {
-    const msg = "Failed to update network policies from api service watch";
+    const msg = "Failed to update network policies from API service watch";
     log.error({ err }, msg);
   }
 }
 
 /**
- * Update the API server CIDR and update the NetworkPolicies
+ * Update the API server CIDR and apply it to the NetworkPolicies.
  *
- * @param slice The EndpointSlice for the API server
- * @param svc The Service for the API server
+ * @param {kind.Service} svc - The Service object representing the Kubernetes API server.
+ * @param {kind.EndpointSlice | string} slice - Either the EndpointSlice for dynamic CIDR generation or a static CIDR string.
  */
-export async function updateAPIServerCIDR(slice: kind.EndpointSlice, svc: kind.Service) {
-  const { endpoints } = slice;
+export async function updateAPIServerCIDR(svc: kind.Service, slice: kind.EndpointSlice | string) {
   const k8sApiIP = svc.spec?.clusterIP;
 
-  // Flatten the endpoints into a list of IPs
-  const peers = endpoints?.flatMap(e => e.addresses);
+  let peers: string[] = [];
 
+  // Handle static CIDR or dynamic EndpointSlice
+  if (typeof slice === "string") {
+    peers.push(slice);
+  } else {
+    const { endpoints } = slice;
+    peers = endpoints?.flatMap(e => `${e.addresses}/32`) || [];
```

Comment:
I might not have the full context here, could `e.addresses` be null causing some null/undefined errors? Not something you introduced. Side question, should we do some form of CIDR validation, i think we discussed this when i originally did some of the remote cidr work.

Reply:
As discussed offline the type here ensures that `endpoints` should be non-optional, and `addresses` is a non-optional field for endpoints, so unless they are just empty (our fallback of `|| []`) we should be okay.